File: OfficeBound.Contracts/Dtos/RequestDto.cs
using OfficeBound.Domain.Enumerations;

namespace OfficeBound.Contracts.Dtos;

public record RequestDto(
    int Id, 
    string Description, 
    RequestType RequestType, 
    DateTime CreatedDate, 
    DateTime RequestDate, 
    RequestStatus RequestStatus, 
    string? RejectionReason,
    int? DepartmentId,
    string? DepartmentName,
    string? CreatedByUsername);

File: OfficeBound.Presentation/Client/officebound/src/models/requestDto.ts
export interface RequestDto {
    id: number | undefined,
    description: string,
    requestType: number,
    createdDate: string | undefined,
    requestDate: string | undefined,
    requestStatus: number,
    rejectionReason: string | null | undefined,
    departmentId: number | null | undefined,
    departmentName: string | null | undefined,
}